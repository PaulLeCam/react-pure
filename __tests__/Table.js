import React from 'react';
import TestRenderer from 'react-test-renderer';

import Table from '../src/Table';

describe('Table', () => {
  it('renders a table with class `pure-table` by default', () => {
    const renderer = TestRenderer.create(<Table className="my-table" />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-table-bordered` class if the `bordered` property is set', () => {
    const renderer = TestRenderer.create(<Table bordered />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-table-horizontal` class if the `horizontal` property is set', () => {
    const renderer = TestRenderer.create(<Table horizontal />);
    expect(renderer.toJSON).toMatchSnapshot();
  });

  it('adds the `pure-table-striped` class if the `striped` property is set to true', () => {
    const renderer = TestRenderer.create(<Table striped />);
    expect(renderer.toJSON).toMatchSnapshot();
  });
});
