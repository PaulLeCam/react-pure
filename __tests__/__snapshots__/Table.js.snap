// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table adds the \`pure-table-bordered\` class if the \`bordered\` property is set 1`] = `[Function]`;

exports[`Table adds the \`pure-table-horizontal\` class if the \`horizontal\` property is set 1`] = `[Function]`;

exports[`Table adds the \`pure-table-striped\` class if the \`striped\` property is set to true 1`] = `[Function]`;

exports[`Table renders a table with class \`pure-table\` by default 1`] = `[Function]`;
